File: repo/solid/ai-translate/webview/pywebview/clipboard.py
import win32gui, pyperclip


class Clipboard:
    def GetActiveWindowTitle(self) -> str:
        # Obter o identificador da janela ativa
        hwnd = win32gui.GetForegroundWindow()
        # Obter o título da janela usando o identificador
        title = win32gui.GetWindowText(hwnd)
        return title


    def GetClipboardText(self) -> str: 
        try: return pyperclip.paste()
        except: return self.GetClipboardText()

File: repo/solid/ai-translate/modules/gowebview/clipboard.ts
export async function GetActiveWindowTitle() { 
    if ('GetActiveWindowTitle' in window) { 
        const response = await window.GetActiveWindowTitle<string[]>()
        if (response?.length) { return response[0] }
    }
}

export async function GetClipboardText() { 
    if ('GetClipboardText' in window) {
        const response = await window.GetClipboardText<string[]>()
        if (response?.length) { return response[0] }
    }
}